core/ui/src/main/java/org/phoebus/ui/docking/DockPane.java

```diff
@@ -243,7 +244,23 @@ private void showContextMenu(final ContextMenuEvent event)
                     ((SplitDock) dock_parent).canMerge())
                 {
                     final MenuItem close = new MenuItem(Messages.DockClose, new ImageView(close_icon));
-                    close.setOnAction(evt -> mergeEmptyAnonymousSplit());
+                    close.setOnAction(evt -> 
+                    {
+                        if (!getName().isBlank())
+                        {
+                            Alert alert = new Alert(Alert.AlertType.CONFIRMATION);
+                            alert.initOwner(dock_parent.getScene().getWindow());
+                            alert.setTitle(Messages.DockCloseNamedPaneTitle);
+                            alert.setContentText(MessageFormat.format(Messages.DockCloseNamedPaneText, getName()));
+                            alert.getButtonTypes().setAll(ButtonType.YES, ButtonType.NO);
+                            alert.showAndWait().ifPresent(type -> {
```

Comment:
Maybe add `DialogHelper.positionDialog(...` so the dialog shows up relative to the dock_parent?
Otherwise, on setups with multiple monitors, the dialog could be way elsewhere.

Reply:
I tried relative to dock_parent, but relative to current DockPane seems better to me...

Reply:
Agree, dock pane is better. In any case, try to assert that the dialog comes up close to the "close" button, not on a different monitor.

Reply:
Good, thanks!